source/cppassist/source/string/conversion.cpp

```diff
@@ -168,5 +168,35 @@ std::u32string encode(const std::string & input, const Encoding encoding)
 //}
 
 
+std::string toLower(const std::string & input)
+{
+    std::string result = input;
+    std::transform(result.begin(), result.end(), result.begin(), ::tolower);
+    return result;
+}
+
+std::string toLower(std::string && input)
+{
+    std::string result;
+    std::swap(result,input);
```

Comment:
Why do you use no ```std::string result = std::move(input);``` here?
Technically it should also be valid to operate directly on the ```input``` (although I would argue to not operate on *in* parameters).